scilog/src/app/core/auth-services/auth.interceptor.ts

```diff
@@ -1,57 +1,85 @@
-import { Injectable } from '@angular/core';
+import { inject, Injectable } from '@angular/core';
 import {
   HttpRequest,
   HttpHandler,
   HttpEvent,
   HttpInterceptor,
   HttpResponse,
-  HttpErrorResponse
+  HttpErrorResponse,
 } from '@angular/common/http';
 import { Observable } from 'rxjs';
 import { tap } from 'rxjs/operators';
+import { ServerSettingsService } from '@shared/config/server-settings.service';
 
 function logout() {
-  localStorage.removeItem("id_token");
+  localStorage.removeItem('id_token');
   localStorage.removeItem('id_session');
+  localStorage.removeItem('scicat_token');
   sessionStorage.removeItem('scilog-auto-selection-logbook');
   location.href = '/login';
-};
-
-function handle_request(handler: HttpHandler, req: HttpRequest<any>) {
-  return handler.handle(req).pipe(tap((event: HttpEvent<any>) => {
-    if (event instanceof HttpResponse) {
-      // console.log(cloned);
-      // console.log("Service Response thr Interceptor");
-    }
-  }, (err: any) => {
-    if (err instanceof HttpErrorResponse) {
-      console.log("err.status", err);
-      if (err.status === 401) {
-        logout();
-      }
-    }
-  }));
 }
 
 @Injectable()
 export class AuthInterceptor implements HttpInterceptor {
+  private serverSettingsService = inject(ServerSettingsService);
 
-  intercept(req: HttpRequest<any>,
-    next: HttpHandler): Observable<HttpEvent<any>> {
+  private isRequestToSciCatBackend(req_url: string): boolean {
+    try {
+      const origin = new URL(req_url).origin;
+      return (
+        origin ===
+        new URL(this.serverSettingsService.getSciCatServerAddress()).origin
+      );
+    } catch (err) {
+      // new URL(...) fails for request to static assets (e.g. /assets/config.json)
+      return false;
+    }
+  }
 
-    const idToken = localStorage.getItem("id_token");
+  private handle_request(handler: HttpHandler, req: HttpRequest<any>) {
```

Comment:
I am not super convinced the login to scicat should be part of the interceptor, rather than invoked at component creation of the scicat widget